### PR TITLE
Feature/cxspa 1498

### PR DESCRIPTION
**Pull request by JedidiahAwuku**

### CODE REVIEW

File: .env-cmdrc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "dev": {
-    "CX_BASE_URL": "https://40.76.109.9:9002"
+    "CX_BASE_URL": "https://api.cg79x9wuu9-eccommerc1-p1-public.model-t.myhybris.cloud/"
```
Comment (SVendittelli) [inline .env-cmdrc]: This should not be permanent

File: feature-libs/pickup-in-store/assets/translations/en/pickup-in-store.i18n.ts [hunk 2]
```diff
@@ -54,6 +54,12 @@ export const storeSearch = {
   viewAllStores: 'View all stores',
 };
 
+/** Translation keys for the SetPreferredStoreComponent */
+export const setPreferredStore = {
+  makeThisMyStore: 'Make This My Store',
+  myStore: 'My Store',
+};
+
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/assets/translations/en/pickup-in-store.i18n.ts]: I believe the rest of these are alphabetical in the file, could this be moved to maintain that?

File: feature-libs/pickup-in-store/assets/translations/en/pickup-in-store.i18n.ts [hunk 3]
```diff
@@ -63,4 +69,5 @@ export const pickupInStore = {
   storeList,
   storeSchedule,
   storeSearch,
+  setPreferredStore,
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/assets/translations/en/pickup-in-store.i18n.ts]: Same as above

File: feature-libs/pickup-in-store/components/container/pickup-option-dialog/pickup-option-dialog.component.spec.ts [hunk 4]
```diff
@@ -132,10 +135,38 @@ describe('PickupOptionDialogComponent', () => {
...
+        guid: 'test',
+        user: { uid: 'anonymous' },
+        code: 'code',
+      })
+    );
+    component.ngOnInit();
+    expect(component.cartId).toEqual('test');
+    expect(component.userId).toEqual('anonymous');
+  });
+  it('ngOnInit should set the cartId and userId for a logged in user', () => {
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/container/pickup-option-dialog/pickup-option-dialog.component.spec.ts]: This is a nitpick, but there should be a blank line between `it` blocks

File: feature-libs/pickup-in-store/components/container/pickup-option-dialog/pickup-option-dialog.component.ts [hunk 5]
```diff
@@ -168,18 +166,26 @@ export class PickupOptionDialogComponent implements OnInit, OnDestroy {
...
-          this.entryNumber,
-          this.userId,
-          preferredStore.name,
-          this.quantity
-        );
-      }
+      return;
     }
+    this.subscription.add(
+      // TODO change this to be their intended store, not preferred store
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/container/pickup-option-dialog/pickup-option-dialog.component.ts]: Remove completed todo comment

File: feature-libs/pickup-in-store/components/container/set-preferred-store/set-preferred-store.component.html [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+how to do this. It will involve a service which interacts with the store.--
+
+Will also need code that when the heart is clicked, sets this store to be the preferred store
+
+The existing Pick up from here button will need some functionality subtracted.It will now only set this store as the
+pickup location for this product. It will not be making this store the preferres store. (Either locally, or via API)
+
+
+
+-->
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/container/set-preferred-store/set-preferred-store.component.html]: What is the purpose of this comment?

File: feature-libs/pickup-in-store/components/container/set-preferred-store/set-preferred-store.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+ * This is a stub of the SetPreferredStoreComponent with the same inputs
+ * for the purposes of testing the components that wrap it.
+ */
+@Component({
+  selector: 'cx-set-preferred-store',
+  template: '',
+})
+export class SetPreferredStoreStubComponent {
+  @Input() pointOfServiceName: PointOfServiceNames;
+}
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/container/set-preferred-store/set-preferred-store.component.spec.ts]: Spec file with no tests?

File: feature-libs/pickup-in-store/components/container/store-list/store-list.component.ts [hunk 8]
```diff
@@ -57,9 +57,6 @@ export class StoreListComponent implements OnInit {
    */
   onSelectStore(store: PointOfServiceStock) {
     const { stockInfo: _, ...pointOfService } = store;
-    const { name = '', displayName = '' } = pointOfService;
-
-    this.preferredStoreService.setPreferredStore({ name, displayName });
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/container/store-list/store-list.component.ts:62]: `preferredStoreService` is no longer used, it should be removed

File: feature-libs/pickup-in-store/components/container/store-list/store-list.component.spec.ts [hunk 9]
```diff
@@ -48,7 +49,9 @@ describe('StoreListComponent', () => {
     fixture = TestBed.createComponent(StoreListComponent);
     component = fixture.componentInstance;
     pickupLocationsSearchService = TestBed.inject(PickupLocationsSearchFacade);
-    preferredStoreService = TestBed.inject(PreferredStoreService);
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/components/container/store-list/store-list.component.spec.ts:51]: `preferredStoreService` is no longer used, it should be removed

File: feature-libs/pickup-in-store/core/facade/intended-pickup-location.service.spec.ts [hunk 10]
```diff
@@ -81,7 +81,11 @@ export class MockIntendedPickupLocationService
   getIntendedLocation(
     _productCode: string
   ): Observable<AugmentedPointOfService | undefined> {
-    return of(undefined);
+    const result: AugmentedPointOfService = {
+      pickupOption: 'pickup' as PickupOption,
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/core/facade/intended-pickup-location.service.spec.ts]: This cast is unnecessary, remove it

File: feature-libs/pickup-in-store/styles/_set-preferred-store.scss [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+  color: var(--cx-color-primary);
+}
+.icon-not-selected {
+  color: var(--cx-color-secondary);
+}
+.setpreferredstore-container {
+  display: flex;
+  margin-top: 0.75rem;
+  font-size: 1rem;
+  font-weight: 600;
```
Comment (SVendittelli) [inline feature-libs/pickup-in-store/styles/_set-preferred-store.scss]: Have a look at the other CSS we've defined. I believe there is a helper method for font size and weight. Talk to me if you can't find it

**Review by SVendittelli — CHANGES_REQUESTED** [review]
Please address these comments and the failing checks

File: .env-cmdrc (hunk 1 above)
Comment (JedidiahAwuku) [reply .env-cmdrc]: Reverted to initial instance

File: feature-libs/pickup-in-store/assets/translations/en/pickup-in-store.i18n.ts (hunk 2 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/assets/translations/en/pickup-in-store.i18n.ts]: Done

File: feature-libs/pickup-in-store/components/container/set-preferred-store/set-preferred-store.component.html (hunk 6 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/container/set-preferred-store/set-preferred-store.component.html]: This served as a guideline for implementation of the feature. It has now been removed.

File: feature-libs/pickup-in-store/components/container/store-list/store-list.component.ts (hunk 8 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/container/store-list/store-list.component.ts:62]: Done

File: feature-libs/pickup-in-store/components/container/store-list/store-list.component.spec.ts (hunk 9 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/container/store-list/store-list.component.spec.ts:51]: Done

File: feature-libs/pickup-in-store/core/facade/intended-pickup-location.service.spec.ts (hunk 10 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/core/facade/intended-pickup-location.service.spec.ts]: Done

File: feature-libs/pickup-in-store/styles/_set-preferred-store.scss (hunk 11 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/styles/_set-preferred-store.scss]: Done

File: feature-libs/pickup-in-store/components/container/set-preferred-store/set-preferred-store.component.spec.ts (hunk 7 above)
Comment (JedidiahAwuku) [reply feature-libs/pickup-in-store/components/container/set-preferred-store/set-preferred-store.component.spec.ts]: Added

**Review by SVendittelli — APPROVED** [review]

**Pull request merged**